pkg/operator/resources/operator/operator_test.go

```diff
@@ -17,75 +19,80 @@ var crdTypeMap = map[string]interface{}{
+var _ = Describe("Operator resource test", func() {
+	It("Test CRD schemas", func() {
+		for crdFileName, cdiObjType := range crdTypeMap {
+
+			schema := getSchema()
+			missingEntries := schema.GetMissingEntries(cdiObjType)
+			for _, missing := range missingEntries {
+				if strings.HasPrefix(missing.Path, "/status") {
+					//Not using subresources, so status is not expected to appear in CRD
+				} else {
+					Fail(fmt.Sprintf("Discrepancy between CRD and Struct"+
+						"Missing or incorrect schema validation at %v, expected type %v  in CRD file %v", missing.Path, missing.Type, crdFileName))
+				}
 			}
 		}
-	}
-}
-
-func TestSampleCustomResources(t *testing.T) {
-
-	var crFileName = "cdi-cr.yaml"
-	root := "./../../../../_out/manifests/release/"
-	schema := getSchema(t)
-	yamlString, err := ioutil.ReadFile(root + crFileName)
-	assert.NoError(t, err, "Error reading %v CR yaml", crFileName)
-	var input map[string]interface{}
-	assert.NoError(t, yaml.Unmarshal([]byte(yamlString), &input))
-	assert.NoError(t, schema.Validate(input), "File %v does not validate against the CRD schema", crFileName)
-}
-
-func TestInvalidCustomResources(t *testing.T) {
-
-	crFileName := []byte(` {
-          "apiVersion":"cdi.kubevirt.io/v1alpha1",
-          "kind":"CDI",
-          "metadata": {
-            "name":"cdi",
-            "namespace":"cdi"
-          },
-          "spec": {
-            "imagePullPolicy":"noValue"
-          }
-        }`)
-	crFileName, err := yaml.JSONToYAML(crFileName)
-	if err != nil {
-		fmt.Printf("err: %v\n", err)
-		return
-	}
+	})
+
+	It("Test sample custom resources", func() {
+		var crFileName = "cdi-cr.yaml"
+		root := "./../../../../_out/manifests/release/"
+		schema := getSchema()
+		yamlString, err := ioutil.ReadFile(root + crFileName)
+		Expect(err).ToNot(HaveOccurred())
+		var input map[string]interface{}
+		err = yaml.Unmarshal([]byte(yamlString), &input)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = schema.Validate(input)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Test invalid custom resources", func() {
+		crFileName := []byte(` {
+		  "apiVersion":"cdi.kubevirt.io/v1alpha1",
+		  "kind":"CDI",
+		  "metadata": {
+		    "name":"cdi",
+		    "namespace":"cdi"
+		  },
+		  "spec": {
+		    "imagePullPolicy":"noValue"
+		  }
+		}`)
+		crFileName, err := yaml.JSONToYAML(crFileName)
+		if err != nil {
+			fmt.Printf("err: %v\n", err)
+			return
+		}
 
-	schema := getSchema(t)
+		schema := getSchema()
 
-	var input map[string]interface{}
-	assert.NoError(t, yaml.Unmarshal([]byte(crFileName), &input))
-	err = schema.Validate(input)
-	if err != nil {
-
-		t.Log(err)
-	}
-	assert.Errorf(t, err, "File %v does not validate against the CRD schema", crFileName)
-}
+		var input map[string]interface{}
+		err = yaml.Unmarshal([]byte(crFileName), &input)
+		Expect(err).ToNot(HaveOccurred())
+		err = schema.Validate(input)
+		By(fmt.Sprintf("err: %d, File %v does not validate against the CRD schema", err, crFileName))
```

Comment:
Missing Expect(err).ToNot(HaveOccurred()), also the By only gets printed when the test fails, if it is the last statement of the test, it will never get printed.